app/controllers/submission_viewing_events_controller.rb

```diff
@@ -1,94 +1,86 @@
+require 'local_storage'
+require 'local_submitted_content'
+
 class SubmissionViewingEventsController < ApplicationController
   include SubmittedContentHelper
   include SubmissionViewingEventHelper
+
+  before_action :ensure_store, only: %i[start_timing end_timing reset_timing hard_save end_round_and_save]
+
   def action_allowed?
     true
   end
 
-  # record time when link or file is opened in new window
-  def record_start_time
-    param_args = params[:submission_viewing_event] # get args from triggering event
-    store = LocalStorage.new() # access local pstore file
-
-    # check if this link is already opened and timed
-    submission_viewing_event_records = store.where(map_id: param_args[:map_id], round: param_args[:round], link: param_args[:link])
-
-    # if opened, end these records with current time
-    if submission_viewing_event_records
-      submission_viewing_event_records.each do |time_record|
-        if time_record.end_at.nil?
-          store.remove(time_record)
-        end
-      end
-    end
-
-    # create new response time record for current link
-    submission_viewing_event = LocalSubmittedContent.new(submission_viewing_event_params)
-    store.save(submission_viewing_event)
+  # Records the start time for a review asset and clears the end time.
+  # The intent here is to signal "we're currently tracking this as being reviewed."
```

Comment:
Mild comment suggestion
```suggestion
  # Records the start time for a review asset and clears the end time.
  # The intent here is to signal "we're currently tracking this while it is being reviewed."
```